src/server/pps/server/api_server.go

```diff
@@ -95,12 +74,62 @@ func (a *apiServer) CreateJob(ctx context.Context, request *ppsclient.CreateJobR
 	if len(repoSet) < len(request.Inputs) {
 		return nil, fmt.Errorf("pachyderm.ppsclient.jobserver: duplicate repo in job")
 	}
+
+	jobID := uuid.NewWithoutDashes()
+
+	var parentJobInfo *persist.JobInfo
+	var err error
+	if request.ParentJob != nil {
+		inspectJobRequest := &ppsclient.InspectJobRequest{Job: request.ParentJob}
+		parentJobInfo, err = a.persistAPIServer.InspectJob(ctx, inspectJobRequest)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	reduce := false
+	if parentJobInfo != nil {
+		for _, jobInput := range parentJobInfo.Inputs {
+			if jobInput.Reduce {
+				reduce = true
+			}
+		}
+	}
```

Comment:
I think `parentJobInfo` is supposed to just be the jobs `JobInfo`. Here's the behavior we want I think:
If `JobInfo.Pipeline` is set use the pipeline repo. Otherwise create a repo just for this job.
If ParentJob is set and Job is not a reduce (not parent job like we're now doing) then the commit we create should use the parent job's output commit as a parent. Otherwise the commit should have no parent.

Double check that to make sure it makes sense to you... I don't entirely trust myself at this point given we were checking the ParentJobInfo.